tst/EnergyPlus/unit/MixedAir.unit.cc

```diff
@@ -824,4 +826,171 @@ namespace EnergyPlus {
+			"    DOAS Relief Air Outlet,  !- Relief Air Stream Node Name",
+			"    DOAS Air Loop Inlet;     !- Return Air Stream Node Name",
+
+			"AirLoopHVAC:ControllerList,",
+			"    DOAS OA System Controllers,  !- Name",
+			"    Controller:OutdoorAir,   !- Controller 1 Object Type",
+			"    DOAS OA Controller;      !- Controller 1 Name",
+		} );
+
+		ASSERT_FALSE( process_idf( idf_objects ) );
+
+		DataGlobals::NumOfTimeStepInHour = 1;
+		DataGlobals::MinutesPerTimeStep = 60 / DataGlobals::NumOfTimeStepInHour;
+		DataGlobals::TimeStep = 1;
+		DataGlobals::HourOfDay = 1;
+		DataEnvironment::DayOfWeek = 1;
+		DataEnvironment::DayOfYear_Schedule = 1;
+		ScheduleManager::UpdateScheduleValues();
+
+		GetOASysInputFlag = true;
+		DataGlobals::BeginEnvrnFlag = true;
+		int AirloopNum = 1;
+		int OASysNum = 1;
+		int HumNum( 1 );
+		int AirInNode( 0 );
+		int AirOutNode( 0 );
+		Real64 WaterConsumptionRate( 0.0 ); // water use rate of the humidifier
+		Real64 ElecPowerInput( 0.0 ); // electric use rate of the humidifier
+
+		AirLoopControlInfo.allocate( AirloopNum ); 
+		AirLoopFlow.allocate( AirloopNum ); 
+		AirLoopFlow( AirloopNum ).DesSupply = 1.0;
+		DataEnvironment::StdRhoAir = 1.2;
+		DataEnvironment::OutBaroPress = 101250.0;
+		DataSizing::SysSizingRunDone = false;
+		DataSizing::CurSysNum = 1;
+
+		GetOutsideAirSysInputs();
+		EXPECT_EQ( 1, NumOASystems );
+		EXPECT_EQ( "DOAS OA SYSTEM", OutsideAirSys( OASysNum ).Name );
+
+		// setup OA system and initialize nodes
+		ManageOutsideAirSystem( OutsideAirSys( OASysNum ).Name, true, AirloopNum, OASysNum );
+		// reset nodes to common property
+		for ( int i = 1; i <= DataLoopNode::NumOfNodes; ++i ) {
+			Node( i ).Temp = 20.0;
+			Node( i ).HumRat = 0.0005;
+			Node( i ).Enthalpy = Psychrometrics::PsyHFnTdbW( DataLoopNode::Node( i ).Temp, DataLoopNode::Node( i ).HumRat );
+			Node( i ).MassFlowRate = 1.0;
+			Node( i ).MassFlowRateMaxAvail = 1.0;
+			Node( i ).Press = 101250.0;
+		}
+		// simulate OA system, common node properties are propagated
+		ManageOutsideAirSystem( OutsideAirSys( OASysNum ).Name, true, AirloopNum, OASysNum );
+		// humidifier water and electric use rate are zero (no Hum Rat setpoint applied)
+		EXPECT_EQ( 0.0, Humidifiers::Humidifier( HumNum ).WaterAdd );
+		EXPECT_EQ( 0.0, Humidifiers::Humidifier( HumNum ).ElecUseRate );
+
+		// Add humidity ratio setpoint to the humidifier air outlet node
+		Node( 2 ).HumRatMin = 0.005;  // humidity ratio setpoint value
+		// simulate OA system
+		ManageOutsideAirSystem( OutsideAirSys( OASysNum ).Name, true, AirloopNum, OASysNum );
+		// get humidifier's air inlet and outlet node number
+		AirInNode = Humidifiers::Humidifier( HumNum ).AirInNode;
+		AirOutNode = Humidifiers::Humidifier( HumNum ).AirOutNode;
+		// Calculate humidifier water consumption rate
+		WaterConsumptionRate = Humidifiers::Humidifier( HumNum ).AirInMassFlowRate * ( DataLoopNode::Node( AirOutNode ).HumRat - DataLoopNode::Node( AirInNode ).HumRat );
```

Comment:
@Nigusse This test would be more robust if you used the specified values ( 0.0005 inlet humrat, 0.005 setpoint and the 1.0 massflowrate) in this equation instead of the values coming back from the humidifier.